### ISSUE TITLE
Angular supports only up to 10 bindings

### ISSUE THREAD

**Issue by maximusk** [angular/angular#15696]
The new angular 4.0.1 version seems to support only up to 10 bindings based [on this source code](https://github.com/angular/angular/blob/master/packages/core/src/view/text.ts#L62) and the following example:

```
@Component({
    moduleId: module.id,
    selector: 'my-app',
    template: `
        <h1>Hello {{name}}</h1>
        <bindings-comp
                [b0]="b0"
                [b1]="b1"
                [b2]="b2"
                [b3]="b3"
                [b4]="b4"
                [b5]="b5"
                [b6]="b6"
                [b7]="b7"
                [b8]="b8"
                [b9]="b9"
                [b10]="b10">
        </bindings-comp>

@Component({
    moduleId: module.id,
    selector: 'bindings-comp',
    template: `{{b0}}{{b1}}{{b2}}{{b3}}{{b4}}{{b5}}{{b6}}{{b7}}{{b8}}{{b9}}{{b10}}`
})
export class BindingsComponent {
    @Input() b0;
    @Input() b1;
    @Input() b2;
    @Input() b3;
    @Input() b4;
    @Input() b5;
    @Input() b6;
    @Input() b7;
    @Input() b8;
    @Input() b9;
    @Input() b10;
}
```

The first 10 are rendered, while the `b10` is not. Is it expected behavior?

**Comment by DzmitryShylovich** [comment]
See https://github.com/angular/angular/issues/12233

**Comment by maximusk** [comment]
Thanks, but it's for the older version of angular (v2). The view generation system was very different there. I'm exploring the new system and just want to confirm what I see in the source code, which is there is no way to render more than 10 bindings

**Comment by DzmitryShylovich** [comment]
Sorry, wrong link. You can have 10+ interpolations http://plnkr.co/edit/uLT4GoBBxtypDdK4Xp5k?p=preview

https://github.com/angular/angular/pull/12710

**Comment by pkozlowski-opensource** [comment]
Yup, this was fixed via #12710 and is available in all Angular versions starting from 2.2.0.

Thnx @DzmitryShylovich for pointing this out!

**Comment by maximusk** [comment]
@DzmitryShylovich, @pkozlowski-opensource , thanks guys, sorry for bothering. I've figured out. It simply uses checkAndUpdateTextDynamic(https://github.com/angular/angular/blob/master/packages/core/src/view/text.ts#L97) function instead of checkAndUpdateTextInline(https://github.com/angular/angular/blob/master/packages/core/src/view/text.ts#L62) when there are more than 10 bindings. I guess most of the time `checkAndUpdateTextDynamic` is used for performance reasons, while the dynamic version `checkAndUpdateTextDynamic` with a loop is used occasionally for many bindings. Can anyone please confirm that?
